Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2113 -- Reflect edited statute dates in frontend

Resolves [EREGCSC-2113](https://jiraent.cms.gov/browse/EREGCSC-2113)

**Description:**

Pulls dates from site configuration in admin panel into front end for Statute Table column header labels.

**This pull request changes:**

- Adds statute table site configuration labels and dates into Statutes template using [best practice](https://adamj.eu/tech/2022/10/06/how-to-safely-pass-data-to-javascript-in-a-django-template/)
- Only shows dates if both label and date exist, and if date is a valid date
- Updates snapshot tests and adds unit front end unit test coverage

**Steps to manually verify this change:**

1. Go into [admin panel for experimental deployment](https://1ghm0p11p3.execute-api.us-east-1.amazonaws.com/dev929/admin/) > Regulations > Site Configuration
2. Edit Date Type and Date
3. Try out a bunch of options.  If Date Type and Date are both valid, the date subtitle should appear in the column header.  If one or both of the Date Type and Date are empty strings or invalid, the date subtitle will not appear.
